[DOCS-12158] Add how throttle works (#32885)
DataDog/documentation: content/en/observability_pipelines/processors/throttle.md

## How the throttle processor works

The throttle processor sets a rate limit on the number of logs sent within a specified time window. While similar to the quota processor, the main difference between the throttle and quota is that the quota processor's time window is fixed at 24 hours and cannot be changed, while the throttle processor's time window can be configured. Since the throttle processor's time window is configurable, the processor has a capacity replenishment rate based on the throttling rate and time window you set. See [Capacity replenishment rate](#capacity-replenishment-rate) for more information.

### Initial capacity

When the throttle processor is enabled, the number of logs the processor allows through immediately is based on the configured **Throttling Rate**. For example, if the **Throttling Rate** is set to `1000` events over 60 seconds:

- And 5,000 events arrive the moment the processor is enabled, the processor allows an initial capacity of 1,000 events to pass through while the rest (4,000 events) are dropped.
- This initial behavior is identical to a quota processor's.

### Capacity replenishment rate

> The throttle processor follows a generic cell rate algorithm, which enables a steady rate of events to pass through. The replenishment  rate is based on the settings of your throttle processor and allows a certain number of events to pass through per second. This rate can be calculated as follows:

Author: ```suggestion
The throttle processor follows a generic cell rate algorithm, which enables a steady rate of events to pass through. The replenishment rate is based on the settings of your throttle processor and allows a certain number of events to pass through per second. This rate can be calculated as follows:
```

## Setup

To set up the processor:

1. Define a [filter query](#filter-query-syntax).
    - Only logs that match the specified filter query are processed.
    - All matched logs get throttled. Logs that are sent within the throttle limit and logs that do not match the filter are sent to the next step. Logs sent after the throttle limit has been reached, are dropped.
1. Set the throttling rate. This is the number of events allowed for a given bucket during the set time window.
    - **Note**: This rate limit is applied on a **per-worker level**. If you scale the number of workers up or down, you may want to adjust the processor rate limit accordingly. You can update the rate limit programmatically using the [Observability Pipelines API][1].
1. Set the time window.
1. Optionally, click **Add Field** if you want to group by a field.

{{% observability_pipelines/processors/filter_syntax %}}

## How the throttle processor works

> The throttle processor sets a rate limit on the number of logs sent within a specified time window. While similar to the quota processor, the main difference between the throttle and quota is that the quota processor's time window is fixed at 24 hours and cannot be changed, while the throttle processor's time window can be configured. Since the throttle processor's time window is configurable, the processor has a capacity replenishment rate based on the throttling rate and time window you set. See [Capacity replenishment rate](#capacity-replenishment-rate) for more information.

Reviewer: Link to quota processor docs on the first mention. Should we capitalize all instances of Quota and Throttle since they are names of a feature? Or do we usually keep lowercase

### Initial capacity

{{< img src="observability_pipelines/processors/throttling_rate.png" alt="The throttle processor with the throttling rate set to 1000 K" style="width:40%;" >}}

When the throttle processor is enabled, the number of logs the processor allows through immediately is based on the configured **Throttling Rate**. For example, if the **Throttling Rate** is set to `1000` events over 60 seconds:

- And 5,000 events arrive the moment the processor is enabled, the processor allows an initial capacity of 1,000 events to pass through while the rest (4,000 events) are dropped.
- This initial behavior is identical to a quota processor's.

### Capacity replenishment rate

The throttle processor uses a generic cell rate algorithm, which enables a steady rate of events to pass through. The replenishment rate is based on the settings of your throttle processor and allows a certain number of events to pass through per second. This rate can be calculated as follows:

> $$\text"Throttle rate" / \text"Time window (in seconds)"$$

Reviewer: In the preview the images of the formulas seems a little broken
<img width="820" height="529" alt="Screenshot 2025-11-21 at 2 17 32 PM" src="https://github.com/user-attachments/assets/26c00de1-0648-4563-a512-dd71b4be7987" />

Author: Weird, I don't see that. Are you using Chrome?

<img width="777" height="453" alt="2025-11-25_13-25-31" src="https://github.com/user-attachments/assets/dec6b221-d9de-4ff1-a670-12a320f1b6e7" />

1. Set the time window.
1. Optionally, click **Add Field** if you want to group by a field.

{{% observability_pipelines/processors/filter_syntax %}}

## How the Throttle processor works

The Throttle processor sets a rate limit on the number of logs sent within a specified time window. While similar to the [Quota processor][2], the main difference between the Throttle and Quota processor is that the Quota processor's time window is fixed at 24 hours and cannot be changed, while the Throttle processor's time window can be configured. Since the Throttle processor's time window is configurable, the processor has a capacity replenishment rate based on the throttling rate and time window you set. See [Capacity replenishment rate](#capacity-replenishment-rate) for more information.

### Initial capacity

{{< img src="observability_pipelines/processors/throttling_rate.png" alt="The Throttle processor with the throttling rate set to 1000 K" style="width:40%;" >}}

> When the Throttle processor is enabled, the number of logs the processor allows through immediately is based on the configured **Throttling Rate**. For example, if the **Throttling Rate** is set to `1000` events over 60 seconds:

Reviewer 2: ```suggestion
When the Throttle processor is enabled, the number of logs the processor allows through immediately is based on the configured **Throttling Rate**. For example, if the **Throttling Rate** is set to `1000` events over 60 seconds, and 5,000 events arrive the moment the processor is enabled:
```

Author: Thanks!

> - And 5,000 events arrive the moment the processor is enabled, the processor allows an initial capacity of 1,000 events to pass through while the rest (4,000 events) are dropped.

Reviewer 2: ```suggestion
- The processor allows an initial capacity of 1,000 events to pass through.
- The remaining 4,000 events are dropped.
```

Reviewer 2: Breaking this bullet up a bit to be clearer about the 2 main outcomes
